Shared feed clusters

caiosba opened the pull request:
## Description

- Delete remaining cluster code we're not using anymore
- Add a rake task to upload and download data from Khousheh
- Simplify cluster code
- Implement API to return cluster data from a feed

References: CV2-4095 and CV2-4172

## How has this been tested?

Automated tests were added for new code.

## Checklist

- [x] I have performed a self-review of my own code
- [x] I have added unit and feature tests, if the PR implements a new feature or otherwise would benefit from additional testing
- [ ] I have added regression tests, if the PR fixes a bug
- [x] I have added logging, exception reporting, and custom tracing with any additional information required for debugging
- [x] I considered secure coding practices when writing this code. Any security concerns are noted above.
- [x] I have commented my code in hard-to-understand areas, if any
- [ ] I have made needed changes to the README
- [x] My changes generate no new warnings
- [ ] If I added a third party module, I included a rationale for doing so and followed our current [guidelines](https://meedan.atlassian.net/wiki/spaces/ENG/overview#Choose-the-%E2%80%9Cright%E2%80%9D-3rd-party-module)

caiosba commented on app/graph/relay_on_rails_schema.rb:
```diff
@@ -13,7 +13,7 @@ class RelayOnRailsSchema < GraphQL::Schema
   lazy_resolve(Concurrent::Future, :value)
 
-  disable_introspection_entry_points
+  disable_introspection_entry_points unless Rails.env.development?
```
FYI @jayjay-w , I had to add this so we're able to update the Relay schema file.

melsawy commented on db/migrate/20240218181609_change_project_media_columns_for_clusters.rb:
```diff
@@ -0,0 +1,6 @@
+class ChangeProjectMediaColumnsForClusters < ActiveRecord::Migration[6.1]
+  def change
+    remove_column :clusters, :project_medias_count
+    add_reference :clusters, :project_media, index: true # Center
```
No need to add this column again as we should remove the line that deleted this column for  `20240212055200_create_cluster_project_medias.rb` migration as column already exists so need need to delete the column and recreate it again

melsawy commented on test/controllers/graphql_controller_5_test.rb:
```diff
@@ -151,22 +151,22 @@ def setup
-    assert_response :success
-    assert_equal 1, JSON.parse(@response.body)['data']['project_media']['cluster']['items']['edges'].size
-  end
+  # TODO: review by Sawy
+  # test "should get cluster information" do
+  #   u = create_user is_admin: true
+  #   f = create_feed
+  #   t = create_team
+  #   f.teams << t
+  #   p = create_project team: t
+  #   pm = create_project_media project: p
+  #   c = create_cluster project_media: pm
+  #   c.project_medias << create_project_media
+  #   authenticate_with_user(u)
+  #   query = 'query { project_media(ids: "' + [pm.id, p.id, t.id].join(',') + '") {  cluster { first_item_at, last_item_at, claim_descriptions(feed_id: ' + f.id.to_s + ') { edges { node { id } } }, items(feed_id: ' + f.id.to_s + ') { edges { node { dbid } } } } } }'
+  #   post :create, params: { query: query, team: t.slug }
+  #   assert_response :success
+  #   assert_equal 1, JSON.parse(@response.body)['data']['project_media']['cluster']['items']['edges'].size
+  # end
```
You can remove this test now as cluster field removes from project_media_type

melsawy reviewed (changes requested):
Great work Dr Caio, just few comments related to migration and tests
please do not forget to run the `check-web` tests against this branch (usually this is your advice)

caiosba commented on app/models/cluster.rb:
```diff
@@ -1,156 +1,21 @@
 class Cluster < ApplicationRecord
   include CheckElasticSearch
```
Good catch!

caiosba commented:
Thanks @melsawy , please re-review - I addressed your review comments. I like that all your comments were related to deleting more code, which is great :D

I also executed integration tests against the branch: https://app.travis-ci.com/github/meedan/check-web/builds/268974604
